tes-adapter/src/main/java/com/epam/pipeline/tesadapter/service/TaskMapper.java

```diff
@@ -65,8 +80,54 @@ public PipelineStart mapToPipelineStart(TesTask tesTask) {
     }
 
     private TesExecutor getExecutorFromTesExecutorsList(List<TesExecutor> tesExecutors) {
-        Assert.notEmpty(tesExecutors, messageHelper.getMessage(MessageConstants.ERROR_PARAMETER_NULL_OR_EMPTY,
-                "executors"));
+        Assert.isTrue(tesExecutors.size() == ONLY_ONE, messageHelper.getMessage(
+                MessageConstants.ERROR_PARAMETER_INCOMPATIBLE_CONTENT, EXECUTORS, tesExecutors));
         return tesExecutors.get(FIRST);
     }
+
+    private String getProperInstanceType(TesTask tesTask, Tool pipelineTool) {
+        Double ramGb = tesTask.getResources().getRamGb();
+        Long cpuCores = tesTask.getResources().getCpuCores();
+        Long toolId = pipelineTool.getId();
+        Long regionId = getProperRegionIdInCloudRegionsByTesZone(tesTask.getResources().getZones());
+        Boolean spot = tesTask.getResources().getPreemptible();
+
+        AllowedInstanceAndPriceTypes allowedInstanceAndPriceTypes = cloudPipelineAPIClient
+                .loadAllowedInstanceAndPriceTypes(toolId, regionId, spot);
+        Assert.isTrue(allowedInstanceAndPriceTypes.getAllowedInstanceTypes() != null, messageHelper.getMessage(
```

Comment:
So if we check that PARAMETER_NULL_OR_EMPTY we should also check that this list is not empty. Try to use some methods from CollectionUtils of ListUtils

Reply:
I replaced with Assert.notEmpty() method that actually checks that collection object not null, and contain at least one element.